sklearn/preprocessing/data.py

```diff
@@ -156,12 +206,14 @@ def scale(X, axis=0, with_mean=True, with_std=True, copy=True):
         if copy:
             X = X.copy()
         _, var = mean_variance_axis(X, axis=0)
-        var = _handle_zeros_in_scale(var)
+        var = _handle_zeros_in_scale(var, copy=False)
```

Comment:
I think I got it. I did not touched this part of the code. But I think you are right. `if with_std` was missing. I wrote a test that broke before and now is passing.